### PR TITLE
50/animation controls

### PR DESCRIPTION
**Pull request by hagoptaminian**
@juancarlosfarah @pyphilia:

- Animation controls are now like in applications 2 and 3.
- Bugs when increasing/decreasing number of lines are fixed. 
- I added a stop button to achieve the effect of stopping an oscillating charge. Unfortunately there is no Material UI icon for a stop button with a circular outline. We could just remove this stop button and its functionality. Not convinced it's adding much.

Let me know what you think and if you encounter any bugs.

### CODE REVIEW

File: src/components/common/AnimationControls.js [hunk 1]
```diff
@@ -26,10 +28,18 @@ import {
...
+  },
+  button: {
+    fontSize: '1.75em',
+  },
   pauseButton: {
     color: yellow[800],
   },
   playButton: { color: green[800] },
+  stopButton: { color: blue[900] },
   resetButton: { color: red[800] },
```
Comment (pyphilia) [inline src/components/common/AnimationControls.js:43]: The colors are different from the 2 other apps. Maybe let's go for 
pause => yellowish-orange
reset => orange
stop => red
See the second app here https://apps.dev.graasp.eu/5acb589d0d5d9464081c2d46/602bdf211db0d51cb392aeb9/latest/index.html

File: src/components/lab/EmittedLine.js [hunk 2]
```diff
@@ -29,24 +30,35 @@ export default class EmittedLine extends Component {
...
     this.beginLineInterval();
   }
 
   componentDidUpdate(prevProps, prevState) {
-    const { isPaused, numberOfLines } = this.props;
-    if (isPaused !== prevProps.isPaused && isPaused) {
+    const { isPaused, oscillation, timerCount, numberOfLines } = this.props;
+    if (isPaused !== prevProps.isPaused && isPaused && !oscillation) {
+      clearInterval(this.emittedLineInterval);
+    } else if (isPaused !== prevProps.isPaused && !isPaused && oscillation) {
```
Comment (pyphilia) [inline src/components/lab/EmittedLine.js:45]: Maybe having `isPaused !== prevProps.isPaused` in a common if-clause would be clearer.

**Review by pyphilia — CHANGES_REQUESTED** [review]
Nice stop, that's really cool. Some comments:
- Disable the stop button when it's not being played.
- I wonder if the reset button shouldn't be moved..? Now that there's a stop button, reset is more about the settings and the simulation. What do you think of putting it in the Sidemenu Header (at the right of Observe)? Or perhaps at the bottom of all settings. Let me know what you think.
- Wasn't it possible to keep the settings while stopping the application? The stop and reset really seem to do the same otherwise.

File: src/components/common/AnimationControls.js [hunk 3]
```diff
@@ -26,10 +28,18 @@ import {
 } from '../../config/constants';
 
 const useStyles = makeStyles(() => ({
+  buttonContainer: {
+    display: 'flex',
+    justifyContent: 'center',
+  },
+  button: {
+    fontSize: '1.75em',
```
Comment (juancarlosfarah) [inline src/components/common/AnimationControls.js:36]: I prefer `rem`, but up to you.

For more on `em` vs `rem`, see:

https://webdesign.tutsplus.com/tutorials/comprehensive-guide-when-to-use-em-vs-rem--cms-23984

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Thanks for this. One minor comment, aside from @pyphilia's.

File: src/components/common/AnimationControls.js (hunk 1 above)
Comment (hagoptaminian) [reply src/components/common/AnimationControls.js:43]: 👍 good catch

File: src/components/common/AnimationControls.js (hunk 3 above)
Comment (hagoptaminian) [reply src/components/common/AnimationControls.js:36]: Very interesting article. Did not know all those nuances. I'm going to leave as `em` for now to be consistent with the other apps/not have to determine the `rem` equivalent of `1.75em`.

**Comment by hagoptaminian** [comment]
@pyphilia good points:

- Disabled the stop button when animation is paused.
- You're right, hitting the stop button should not reset other settings. I fixed this. And you're right that the stop button is quite similar to the reset button, with the main difference being that the reset button returns the charge to its origin, while the stop button stops it in place.
- Interesting point on moving the reset button. On the one hand, you're right that Reset does not technically need to be part of the animation controls. On the other, it makes sense to have these Material icons together in one area, and it might look a bit weird to have an icon floating elsewhere. Of course if we move the button here it would make sense to do it for the other apps too. Let's discuss on Friday...

**Comment by hagoptaminian** [comment]
@juancarlosfarah @pyphilia going to go ahead and merge this

**Review by juancarlosfarah — APPROVED** [review]
🥇

**Pull request merged**